Make TmpNameAllArchs obsolete by enhancing TmpName

fingolfin opened the pull request:
Back in 2011, a new library function `TmpNameAllArchs` was added which wraps the kernel function `TmpName` and adds some Windows specific sauce to it. This PR enhances `TmpName` so that `TmpNameAllArchs` can be turned into a simple synonym for `TmpName`. It does so by aligning the code of `FuncTmpName` and `FuncTmpDirectory` more closely. (They still differ in that the latter honors the the `TMPDIR` env var, the former does not; we might want to change that, but I didn't want to make too big a change in one go.

There is another difference between the old and new code: the old used `C:/WINDOWS/Temp/`, the new uses `/cygdrive/c/WINDOWS/Temp/`. I *hope* that doesn't matter or even is an improvement, *but* it would be really good if somebody with Windows access could verify that (beyond what little the tests in our test suite do).

Resolves #3328

The other change in this PR is the removal of the library function `ShortFileNameWindows` which @hulpke added in 2009 but which hasn't ever been used by GAP or any package, as far as I can tell. It's also undocumented, and it seems not so useful to access 8.3 filenames these days. But perhaps I am mistaken, and there are usecases (in particular, I have no idea what the motivation for adding it back then was).

ChrisJefferson commented:
windows vs cygwin names is a pain. If you pass cygwin filenames to window executables, they will get confused, while cygwin executables will accept windows paths, so there is an argument windows paths are better. But we are inconsistent with what types of paths we give out all over the place.

fingolfin commented:
@ChrisJefferson I am of course happy to use a `C:\...` style path here as well. Main thing is that we really need somebody to test the code in this PR on a Windows machine, and see if any packages are affected by it. I am not in any position to do that, so I am hoping that somebody can help with it; be it @alex-konovalov or perhaps @pedritomelenas or somebody else.

On the long run, we really need better Windows testing, too. If "somebody" could figure out how to setup package tests similar to https://travis-ci.org/gap-infra/gap-docker-pkg-tests-master but using Windows, that'd be awesome (regardless of whether it is based on AppVeyor (which we already use for Windows testing, but which I find otherwise rather clunky to use), [Travis for Windows](https://help.github.com/en/articles/setting-up-continuous-integration-using-github-actions), [Azure Pipelines](https://azure.microsoft.com/de-de/services/devops/pipelines/), [GitHub Actions](https://help.github.com/en/articles/setting-up-continuous-integration-using-github-actions), [CircleCI](https://azure.microsoft.com/de-de/services/devops/pipelines/), or whatever.... They all offer free Windows CI for open source projects in one way or another.

fingolfin commented:
I guess issue #2023 is related...

pedritomelenas commented:
On `gap.sh` I get the following (just a warning from crisp)
```
 ┌───────┐   GAP 4.dev of today
 │  GAP  │   https://www.gap-system.org
 └───────┘   Architecture: x86_64-unknown-cygwin-default64-kv7
 Configuration:  gmp 6.1.2, GASMAN, readline
 Loading the library and packages ...
#I  DeclareGlobalFunction: too many arguments in /home/pedro/tmp/gap-old/pkg/crisp-1.4.4/lib/util.gd:16
 Packages:   AClib 1.3.1, Alnuth 3.1.1, AtlasRep 2.1.0, AutoDoc 2019.09.04,
             AutPGrp 1.10.1, CRISP 1.4.4, Cryst 4.1.20, CrystCat 1.1.9,
             CTblLib 1.2.2, FactInt 1.6.2, FGA 1.4.0, Forms 1.2.5,
             GAPDoc 1.6.3, genss 1.6.6, IO 4.7.0, IRREDSOL 1.4, LAGUNA 3.9.3,
             orb 4.8.3, Polenta 1.3.9, Polycyclic 2.15.1, PrimGrp 3.3.2,
             RadiRoot 2.8, recog 1.3.2, ResClasses 4.7.2, SmallGrp 1.4.1,
             Sophus 1.24, SpinSym 1.5.2, TomLib 1.2.8, TransGrp 2.0.4,
             utils 0.67
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
gap> dirs := [
>   DirectoriesLibrary( "tst/testinstall" ),
];
TestDirectory( dirs, rec(exitGAP := false) );> ];
[ [ dir("/home/pedro/tmp/gap-old/tst/testinstall/") ] ]
gap> TestDirectory( dirs, rec(exitGAP := false) );
Architecture: x86_64-unknown-cygwin-default64-kv7

testing: /home/pedro/tmp/gap-old/tst/testinstall/ConjNatSym.tst
      78 ms (0 ms GC) and 23.8MB allocated for ConjNatSym.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/DirectProductElement.tst
      63 ms (63 ms GC) and 667KB allocated for DirectProductElement.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ElmsBlist.tst
      78 ms (62 ms GC) and 4.33MB allocated for ElmsBlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/ConcatenationOfVect\
ors.tst
      47 ms (47 ms GC) and 822KB allocated for MatrixObj/ConcatenationOfVector\
s.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/CopySubVector.tst
      62 ms (47 ms GC) and 56.6KB allocated for MatrixObj/CopySubVector.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/DistanceOfVectors.t\
st
      47 ms (47 ms GC) and 69.3KB allocated for MatrixObj/DistanceOfVectors.ts\
t
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/ExtractSubvector.ts\
t
      47 ms (47 ms GC) and 44.2KB allocated for MatrixObj/ExtractSubvector.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/IdentityMatrix.tst
      47 ms (47 ms GC) and 1.64MB allocated for MatrixObj/IdentityMatrix.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/ListOp.tst
      62 ms (62 ms GC) and 81.3KB allocated for MatrixObj/ListOp.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/Matrix.tst
      32 ms (32 ms GC) and 118KB allocated for MatrixObj/Matrix.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/PositionNonZero.tst
      47 ms (47 ms GC) and 86.2KB allocated for MatrixObj/PositionNonZero.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/Randomize.tst
      46 ms (31 ms GC) and 414KB allocated for MatrixObj/Randomize.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/TraceMat.tst
      47 ms (47 ms GC) and 800KB allocated for MatrixObj/TraceMat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/Unpack.tst
      47 ms (47 ms GC) and 61.9KB allocated for MatrixObj/Unpack.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/WeightOfVector.tst
      47 ms (47 ms GC) and 69.1KB allocated for MatrixObj/WeightOfVector.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/MatrixObj/ZeroVector.tst
      31 ms (31 ms GC) and 58.4KB allocated for MatrixObj/ZeroVector.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/alghom.tst
      63 ms (47 ms GC) and 2.90MB allocated for alghom.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/algmat.tst
    1047 ms (94 ms GC) and 220MB allocated for algmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/algrep.tst
    1265 ms (93 ms GC) and 247MB allocated for algrep.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/algsc.tst
     375 ms (94 ms GC) and 30.8MB allocated for algsc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/associate.tst
     641 ms (79 ms GC) and 88.2MB allocated for associate.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/attribute.tst
      62 ms (62 ms GC) and 326KB allocated for attribute.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/auto.tst
       0 ms (0 ms GC) and 66.8KB allocated for auto.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/bitfields.tst
      63 ms (47 ms GC) and 187KB allocated for bitfields.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/boolean.tst
      47 ms (47 ms GC) and 360KB allocated for boolean.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/bound.tst
      47 ms (47 ms GC) and 198KB allocated for bound.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/break.tst
      47 ms (47 ms GC) and 322KB allocated for break.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/callfunc.tst
      46 ms (46 ms GC) and 1.46MB allocated for callfunc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/coder.tst
      63 ms (63 ms GC) and 538KB allocated for coder.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/coding.tst
      47 ms (47 ms GC) and 289KB allocated for coding.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/coll.tst
      62 ms (47 ms GC) and 3.33MB allocated for coll.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/combinat.tst
     110 ms (63 ms GC) and 14.5MB allocated for combinat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/constant.tst
       0 ms (0 ms GC) and 280KB allocated for constant.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/constructor.tst
       0 ms (0 ms GC) and 71.5KB allocated for constructor.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ctbl.tst
     203 ms (47 ms GC) and 18.6MB allocated for ctbl.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ctblfuns.tst
     234 ms (94 ms GC) and 30.6MB allocated for ctblfuns.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ctblmoli.tst
     703 ms (126 ms GC) and 86.2MB allocated for ctblmoli.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ctblmono.tst
    2500 ms (314 ms GC) and 332MB allocated for ctblmono.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ctblsolv.tst
     250 ms (79 ms GC) and 27.4MB allocated for ctblsolv.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/cyclotom.tst
     344 ms (78 ms GC) and 12.1MB allocated for cyclotom.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/declarefunction.tst
      63 ms (63 ms GC) and 132KB allocated for declarefunction.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/depth.tst
      46 ms (46 ms GC) and 81.3KB allocated for depth.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/dict.tst
      63 ms (63 ms GC) and 210KB allocated for dict.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/dir.tst
      62 ms (62 ms GC) and 100KB allocated for dir.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/domain.tst
      63 ms (63 ms GC) and 111KB allocated for domain.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/dt.tst
     391 ms (62 ms GC) and 58.3MB allocated for dt.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/eigen.tst
      78 ms (78 ms GC) and 167KB allocated for eigen.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/error.tst
       0 ms (0 ms GC) and 63.7KB allocated for error.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/euclidean.tst
    1500 ms (62 ms GC) and 84.1MB allocated for euclidean.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ffe.tst
    1515 ms (78 ms GC) and 31.9MB allocated for ffe.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ffeconway.tst
     360 ms (78 ms GC) and 12.9MB allocated for ffeconway.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/files/files.tst
      78 ms (78 ms GC) and 943KB allocated for files/files.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/fldabnum.tst
     109 ms (62 ms GC) and 6.26MB allocated for fldabnum.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/float.tst
      94 ms (78 ms GC) and 1.39MB allocated for float.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/flush.tst
      62 ms (62 ms GC) and 105KB allocated for flush.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/format.tst
      47 ms (47 ms GC) and 256KB allocated for format.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/fpmon.tst
      63 ms (63 ms GC) and 617KB allocated for fpmon.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/function.tst
      78 ms (78 ms GC) and 657KB allocated for function.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/gaussian.tst
      62 ms (62 ms GC) and 166KB allocated for gaussian.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/gprd.tst
      47 ms (47 ms GC) and 543KB allocated for gprd.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/gprdmat.tst
     141 ms (78 ms GC) and 7.62MB allocated for gprdmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/basic.tst
     594 ms (92 ms GC) and 59.9MB allocated for grp/basic.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/clasmax.tst
      15 ms (0 ms GC) and 1.95MB allocated for grp/clasmax.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/classic-G.tst
     625 ms (124 ms GC) and 74.7MB allocated for grp/classic-G.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/classic-PG.tst
     110 ms (78 ms GC) and 2.33MB allocated for grp/classic-PG.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/classic-PS.tst
      78 ms (62 ms GC) and 1007KB allocated for grp/classic-PS.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/classic-S.tst
     437 ms (62 ms GC) and 41.3MB allocated for grp/classic-S.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/classic-forms.tst
     922 ms (109 ms GC) and 167MB allocated for grp/classic-forms.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/glzmodmz.tst
     391 ms (78 ms GC) and 38.7MB allocated for grp/glzmodmz.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/imf.tst
       0 ms (0 ms GC) and 213KB allocated for grp/imf.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/perf.tst
     781 ms (110 ms GC) and 120MB allocated for grp/perf.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/ree.tst
     109 ms (78 ms GC) and 2.58MB allocated for grp/ree.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grp/suzuki.tst
      79 ms (79 ms GC) and 231KB allocated for grp/suzuki.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grpfp.tst
    1109 ms (126 ms GC) and 116MB allocated for grpfp.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grpfree.tst
      94 ms (78 ms GC) and 1.78MB allocated for grpfree.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grpmat.tst
    1422 ms (469 ms GC) and 467MB allocated for grpmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grppc.tst
     328 ms (78 ms GC) and 29.1MB allocated for grppc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/grpperm.tst
    5031 ms (514 ms GC) and 895MB allocated for grpperm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hashkeybag.tst
      94 ms (94 ms GC) and 89.3KB allocated for hashkeybag.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/help.tst
       0 ms (0 ms GC) and 62.7KB allocated for help.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/atomic_basic.tst
      78 ms (78 ms GC) and 253KB allocated for hpc/atomic_basic.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/atomic_compare.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 77.7KB allocated for hpc/atomic_compare.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/atomic_list.tst
      62 ms (62 ms GC) and 266KB allocated for hpc/atomic_list.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/atomic_list_hpc.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 41.9KB allocated for hpc/atomic_list_hpc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/channels.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 35.4KB allocated for hpc/channels.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/comprvec.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 50.0KB allocated for hpc/comprvec.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/demo.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 12.3KB allocated for hpc/demo.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/fix-coverage.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 17.5KB allocated for hpc/fix-coverage.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/queue.tst
#I  Test: File does not contain any tests!
      16 ms (0 ms GC) and 60.2KB allocated for hpc/queue.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/serialize.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 104KB allocated for hpc/serialize.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/stdtasks.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 50.0KB allocated for hpc/stdtasks.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/tasks.tst
      78 ms (62 ms GC) and 873KB allocated for hpc/tasks.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/hpc/threads.tst
      94 ms (94 ms GC) and 271KB allocated for hpc/threads.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/immutable.tst
      62 ms (62 ms GC) and 131KB allocated for immutable.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/infinity.tst
      78 ms (62 ms GC) and 165KB allocated for infinity.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/info.tst
      79 ms (79 ms GC) and 216KB allocated for info.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/intarith.tst
    1578 ms (108 ms GC) and 72.8MB allocated for intarith.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/integer.tst
     609 ms (125 ms GC) and 207MB allocated for integer.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/interpreter.tst
     547 ms (94 ms GC) and 43.3MB allocated for interpreter.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/invsgp.tst
     109 ms (94 ms GC) and 2.21MB allocated for invsgp.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kbsemi.tst
     110 ms (94 ms GC) and 911KB allocated for kbsemi.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/ariths.tst
     109 ms (93 ms GC) and 688KB allocated for kernel/ariths.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/blister.tst
     125 ms (109 ms GC) and 2.21MB allocated for kernel/blister.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/bool.tst
      94 ms (94 ms GC) and 108KB allocated for kernel/bool.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/calls.tst
      94 ms (79 ms GC) and 306KB allocated for kernel/calls.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/exprs.tst
      93 ms (93 ms GC) and 216KB allocated for kernel/exprs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/funcs.tst
       0 ms (0 ms GC) and 7.58KB allocated for kernel/funcs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/gap.tst
      16 ms (0 ms GC) and 315KB allocated for kernel/gap.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/integer.tst
      78 ms (78 ms GC) and 230KB allocated for kernel/integer.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/intfuncs.tst
      78 ms (78 ms GC) and 53.3KB allocated for kernel/intfuncs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/intrprtr.tst
      63 ms (63 ms GC) and 58.3KB allocated for kernel/intrprtr.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/listfunc.tst
      78 ms (78 ms GC) and 153KB allocated for kernel/listfunc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/lists.tst
      78 ms (78 ms GC) and 285KB allocated for kernel/lists.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/macfloat.tst
      78 ms (78 ms GC) and 76.3KB allocated for kernel/macfloat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/object.tst
      78 ms (78 ms GC) and 168KB allocated for kernel/object.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/objfgelm.tst
      63 ms (63 ms GC) and 47.9KB allocated for kernel/objfgelm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/opers.tst
      78 ms (78 ms GC) and 595KB allocated for kernel/opers.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/range.tst
      78 ms (78 ms GC) and 160KB allocated for kernel/range.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/read.tst
      78 ms (62 ms GC) and 370KB allocated for kernel/read.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/records.tst
      78 ms (78 ms GC) and 191KB allocated for kernel/records.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/scanner.tst
      94 ms (94 ms GC) and 259KB allocated for kernel/scanner.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/sctable.tst
      78 ms (78 ms GC) and 94.7KB allocated for kernel/sctable.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/set.tst
      63 ms (63 ms GC) and 126KB allocated for kernel/set.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/stats.tst
      78 ms (78 ms GC) and 195KB allocated for kernel/stats.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/streams.tst
      62 ms (62 ms GC) and 302KB allocated for kernel/streams.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/kernel/stringobj.tst
      78 ms (78 ms GC) and 177KB allocated for kernel/stringobj.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/linecontinuation.tst
      63 ms (63 ms GC) and 90.5KB allocated for linecontinuation.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/list.tst
      78 ms (62 ms GC) and 1012KB allocated for list.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/listgen.tst
      94 ms (78 ms GC) and 754KB allocated for listgen.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/listindex.tst
     109 ms (93 ms GC) and 659KB allocated for listindex.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/localvars.tst
       0 ms (0 ms GC) and 122KB allocated for localvars.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/longnumber.tst
      94 ms (94 ms GC) and 1.05MB allocated for longnumber.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/mapphomo.tst
      78 ms (62 ms GC) and 2.35MB allocated for mapphomo.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/mapping.tst
     469 ms (110 ms GC) and 32.1MB allocated for mapping.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/matblock.tst
     109 ms (109 ms GC) and 408KB allocated for matblock.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/matrix.tst
       0 ms (0 ms GC) and 122KB allocated for matrix.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/meataxe.tst
     828 ms (530 ms GC) and 249MB allocated for meataxe.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/memoize.tst
     125 ms (125 ms GC) and 162KB allocated for memoize.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/method-rankfns.tst
     156 ms (94 ms GC) and 5.88MB allocated for method-rankfns.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/method-reordering.tst
     407 ms (125 ms GC) and 44.2MB allocated for method-reordering.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/method.tst
      93 ms (93 ms GC) and 147KB allocated for method.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/methwhy.tst
      63 ms (63 ms GC) and 370KB allocated for methwhy.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/mgmring.tst
      62 ms (62 ms GC) and 297KB allocated for mgmring.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/misc.tst
       0 ms (0 ms GC) and 185KB allocated for misc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/modfree.tst
     125 ms (94 ms GC) and 1.93MB allocated for modfree.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/modifiers.tst
     110 ms (94 ms GC) and 95.7KB allocated for modifiers.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/monofree.tst
      78 ms (78 ms GC) and 574KB allocated for monofree.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/morpheus.tst
     375 ms (94 ms GC) and 42.7MB allocated for morpheus.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/nanoseconds.tst
     109 ms (109 ms GC) and 54.2KB allocated for nanoseconds.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/numtheor.tst
     125 ms (63 ms GC) and 11.5MB allocated for numtheor.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/objlist.tst
       0 ms (0 ms GC) and 196KB allocated for objlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/objmap.tst
     110 ms (78 ms GC) and 275KB allocated for objmap.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/objset.tst
     406 ms (62 ms GC) and 17.0MB allocated for objset.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/obsolete.tst
      78 ms (78 ms GC) and 117KB allocated for obsolete.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/onecohom.tst
     313 ms (94 ms GC) and 34.6MB allocated for onecohom.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/oper1.tst
      15 ms (0 ms GC) and 281KB allocated for oper1.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/AutomorphismGroup.tst
     360 ms (110 ms GC) and 33.7MB allocated for opers/AutomorphismGroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/BindingsOfClosure.tst
     125 ms (125 ms GC) and 114KB allocated for opers/BindingsOfClosure.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/CyclesFromList.tst
      93 ms (93 ms GC) and 66.1KB allocated for opers/CyclesFromList.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/CyclotomicField.tst
      78 ms (78 ms GC) and 173KB allocated for opers/CyclotomicField.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/EmptyPlist.tst
      94 ms (94 ms GC) and 64.1KB allocated for opers/EmptyPlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/FittingSubgroup.tst
     156 ms (78 ms GC) and 14.5MB allocated for opers/FittingSubgroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/FrattiniSubgroup.tst
     250 ms (94 ms GC) and 26.7MB allocated for opers/FrattiniSubgroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/HallSubgroup.tst
     766 ms (125 ms GC) and 129MB allocated for opers/HallSubgroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/HexStringBlist.tst
     125 ms (109 ms GC) and 794KB allocated for opers/HexStringBlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/InverseMatMod.tst
      94 ms (94 ms GC) and 366KB allocated for opers/InverseMatMod.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/IsCentral.tst
      94 ms (62 ms GC) and 2.85MB allocated for opers/IsCentral.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/IsPNilpotent.tst
      93 ms (93 ms GC) and 311KB allocated for opers/IsPNilpotent.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/IsSolvableGroup.tst
     219 ms (78 ms GC) and 24.1MB allocated for opers/IsSolvableGroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/LatticeByCyclicExtensio\
n.tst
     188 ms (109 ms GC) and 18.5MB allocated for opers/LatticeByCyclicExtensio\
n.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/ListBlist.tst
      93 ms (93 ms GC) and 49.1KB allocated for opers/ListBlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/LocationFunc.tst
      63 ms (63 ms GC) and 56.2KB allocated for opers/LocationFunc.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/MaximalNormalSubgroups.\
tst
     312 ms (94 ms GC) and 30.9MB allocated for opers/MaximalNormalSubgroups.t\
st
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/MemoryUsage.tst
      78 ms (78 ms GC) and 261KB allocated for opers/MemoryUsage.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/MinimalNormalSubgroups.\
tst
    1422 ms (185 ms GC) and 231MB allocated for opers/MinimalNormalSubgroups.t\
st
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/NormalHallSubgroups.tst
     750 ms (141 ms GC) and 125MB allocated for opers/NormalHallSubgroups.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/NormalSubgroups.tst
     157 ms (125 ms GC) and 4.29MB allocated for opers/NormalSubgroups.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/Quotient.tst
      93 ms (93 ms GC) and 363KB allocated for opers/Quotient.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/RandomInvertibleMat.tst
     172 ms (94 ms GC) and 8.65MB allocated for opers/RandomInvertibleMat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/RandomMat.tst
     125 ms (94 ms GC) and 4.40MB allocated for opers/RandomMat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/RandomUnimodularMat.tst
     110 ms (78 ms GC) and 5.41MB allocated for opers/RandomUnimodularMat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/SimpleGroup.tst
    1171 ms (234 ms GC) and 199MB allocated for opers/SimpleGroup.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/Socle.tst
     454 ms (110 ms GC) and 56.4MB allocated for opers/Socle.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/StructuralCopy.tst
     125 ms (125 ms GC) and 55.3KB allocated for opers/StructuralCopy.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/StructureDescription.ts\
t
    2765 ms (218 ms GC) and 439MB allocated for opers/StructureDescription.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/SubdirectProduct.tst
     156 ms (125 ms GC) and 3.12MB allocated for opers/SubdirectProduct.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/SubdirectProducts.tst
     829 ms (94 ms GC) and 66.9MB allocated for opers/SubdirectProducts.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/SylowSystem.tst
     125 ms (125 ms GC) and 213KB allocated for opers/SylowSystem.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/opers/TriangulizedMat.tst
     109 ms (109 ms GC) and 57.0KB allocated for opers/TriangulizedMat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/oprt.tst
      94 ms (94 ms GC) and 514KB allocated for oprt.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/package.tst
     359 ms (125 ms GC) and 18.8MB allocated for package.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/perm.tst
     688 ms (406 ms GC) and 653MB allocated for perm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/pgroups.tst
     281 ms (125 ms GC) and 29.0MB allocated for pgroups.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/pperm.tst
    1719 ms (265 ms GC) and 418MB allocated for pperm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/pragma.tst
       0 ms (0 ms GC) and 37.9KB allocated for pragma.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/randlist.tst
     109 ms (109 ms GC) and 402KB allocated for randlist.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/random.tst
    2781 ms (188 ms GC) and 293MB allocated for random.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/range.tst
     203 ms (125 ms GC) and 17.1MB allocated for range.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ratfun.tst
    1719 ms (125 ms GC) and 96.8MB allocated for ratfun.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/ratfun_gf5.tst
     484 ms (141 ms GC) and 35.1MB allocated for ratfun_gf5.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/rationals.tst
       0 ms (0 ms GC) and 88.6KB allocated for rationals.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/read.tst
     141 ms (125 ms GC) and 295KB allocated for read.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/recordname.tst
     125 ms (125 ms GC) and 382KB allocated for recordname.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/reesmat.tst
     141 ms (78 ms GC) and 11.5MB allocated for reesmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/relation.tst
     156 ms (125 ms GC) and 1.06MB allocated for relation.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/restrictedperm.tst
       0 ms (0 ms GC) and 3.04MB allocated for restrictedperm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/rwspcgrp.tst
     453 ms (141 ms GC) and 70.7MB allocated for rwspcgrp.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/rwspcsng.tst
     516 ms (171 ms GC) and 132MB allocated for rwspcsng.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/semicong.tst
     125 ms (93 ms GC) and 6.10MB allocated for semicong.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/semigrp.tst
     187 ms (125 ms GC) and 9.00MB allocated for semigrp.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/semipperm.tst
     172 ms (125 ms GC) and 7.99MB allocated for semipperm.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/semirel.tst
     141 ms (109 ms GC) and 6.11MB allocated for semirel.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/semitran.tst
     218 ms (125 ms GC) and 10.5MB allocated for semitran.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/set.tst
     141 ms (110 ms GC) and 4.39MB allocated for set.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/small_groups.tst
     203 ms (109 ms GC) and 13.6MB allocated for small_groups.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/stabchain.tst
     641 ms (110 ms GC) and 33.4MB allocated for stabchain.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/startendwith.tst
     109 ms (109 ms GC) and 85.5KB allocated for startendwith.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/straight.tst
      94 ms (94 ms GC) and 233KB allocated for straight.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/streams.tst
      94 ms (94 ms GC) and 427KB allocated for streams.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/stringobj.tst
     453 ms (94 ms GC) and 14.7MB allocated for stringobj.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/strings.tst
     109 ms (109 ms GC) and 247KB allocated for strings.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/switch.tst
     188 ms (188 ms GC) and 132KB allocated for switch.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/sylowhall.tst
    1031 ms (265 ms GC) and 366MB allocated for sylowhall.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/syntaxtree.tst
     875 ms (47 ms GC) and 251MB allocated for syntaxtree.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/testing.tst
     109 ms (109 ms GC) and 50.2KB allocated for testing.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/testunix/streamio.tst
     110 ms (110 ms GC) and 79.4KB allocated for testunix/streamio.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/testunix/streams.tst
      93 ms (93 ms GC) and 39.0KB allocated for testunix/streams.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/tilde.tst
     125 ms (94 ms GC) and 459KB allocated for tilde.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/trace.tst
     110 ms (110 ms GC) and 130KB allocated for trace.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/trans.tst
    1265 ms (645 ms GC) and 338MB allocated for trans.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/triviso.tst
     125 ms (125 ms GC) and 65.0KB allocated for triviso.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/tuples.tst
     125 ms (125 ms GC) and 45.2KB allocated for tuples.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/type.tst
       0 ms (0 ms GC) and 193KB allocated for type.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/unbound.tst
      94 ms (94 ms GC) and 160KB allocated for unbound.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/unknown.tst
      94 ms (94 ms GC) and 120KB allocated for unknown.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/varargs.tst
     109 ms (93 ms GC) and 189KB allocated for varargs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/vecmat.tst
     282 ms (110 ms GC) and 5.90MB allocated for vecmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/vspchom.tst
     171 ms (125 ms GC) and 3.34MB allocated for vspchom.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/vspcmali.tst
     172 ms (125 ms GC) and 8.54MB allocated for vspcmali.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/vspcmat.tst
     172 ms (109 ms GC) and 8.19MB allocated for vspcmat.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/vspcrow.tst
     235 ms (125 ms GC) and 41.7MB allocated for vspcrow.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/weakptr-badargs.tst
     125 ms (125 ms GC) and 102KB allocated for weakptr-badargs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/weakptr.tst
     312 ms (296 ms GC) and 16.0MB allocated for weakptr.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/wordrep.tst
     219 ms (109 ms GC) and 56.3MB allocated for wordrep.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/xfuncs.tst
     125 ms (109 ms GC) and 643KB allocated for xfuncs.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/xgap.tst
     437 ms (94 ms GC) and 55.3MB allocated for xgap.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/zlattice.tst
     141 ms (141 ms GC) and 48.6KB allocated for zlattice.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/zmodnz.tst
     125 ms (109 ms GC) and 2.27MB allocated for zmodnz.tst
testing: /home/pedro/tmp/gap-old/tst/testinstall/zmodnze.tst
     125 ms (125 ms GC) and 1.28MB allocated for zmodnze.tst
-----------------------------------
total     67844 ms (22726 ms GC) and 8.44GB allocated
              0 failures in 249 files

#I  No errors detected while testing

true
```
I will continue with the rest of tests.

pedritomelenas commented:
Got some issues on testextra
```
gap> TestDirectory([DirectoriesLibrary("tst/testextra"),],rec(exitGAP:=false));
Architecture: x86_64-unknown-cygwin-default64-kv7

testing: /home/pedro/tmp/gap-old/tst/testextra/ctbl.tst
 1359641 ms (97632 ms GC) and 250GB allocated for ctbl.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/ctblpope.tst
  314094 ms (4179 ms GC) and 18.6GB allocated for ctblpope.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/grpauto.tst
 1064734 ms (109203 ms GC) and 373GB allocated for grpauto.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/grplatt.tst
   68875 ms (9668 ms GC) and 19.9GB allocated for grplatt.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/grpperm.tst
########> Diff in /home/pedro/tmp/gap-old/tst/testextra/grpperm.tst:260
# Input is:
Length(coh.cohomology);
# Expected output:
2
# But found:
1
########
########> Diff in /home/pedro/tmp/gap-old/tst/testextra/grpperm.tst:264
# Input is:
Length(p);
# Expected output:
4
# But found:
2
########
########> Diff in /home/pedro/tmp/gap-old/tst/testextra/grpperm.tst:267
# Input is:
List(p,Size);
# Expected output:
[ 61440, 61440, 61440, 61440 ]
# But found:
[ 61440, 61440 ]
########
  387375 ms (186883 ms GC) and 98.8GB allocated for grpperm.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/helpsys.tst
  515906 ms (57294 ms GC) and 67.1GB allocated for helpsys.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/objmap.tst
   10156 ms (529 ms GC) and 1.21GB allocated for objmap.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/objset.tst
    8157 ms (392 ms GC) and 390MB allocated for objset.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/perfect.tst
   52453 ms (1738 ms GC) and 2.78GB allocated for perfect.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/small_groups2.tst
   32015 ms (3812 ms GC) and 5.16GB allocated for small_groups2.tst
testing: /home/pedro/tmp/gap-old/tst/testextra/switch_obj.tst
  229516 ms (229485 ms GC) and 101MB allocated for switch_obj.tst
-----------------------------------
total   4042922 ms (700815 ms GC) and 837GB allocated
              3 failures in 1 of 11 files

#I  Errors detected while testing

false
```

alex-konovalov commented:
I see the same errors in the master, so they should not be relevant to this PR. They may be of an interest to @hulpke - is this something relevant to #3707 by any chance?

hulpke commented:
@alex-konovalov No, nothing related to #3707. I think what happened is that the two modules of dimension 4 changed positions.

pedritomelenas commented:
And the last one. 
```
gap> TestDirectory([DirectoriesLibrary("tst/teststandard"),],rec(exitGAP:=false));
Architecture: x86_64-unknown-cygwin-default64-kv7

testing: /home/pedro/tmp/gap-old/tst/teststandard/algext.tst
     125 ms (63 ms GC) and 8.18MB allocated for algext.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/arithlst.tst
   26282 ms (887 ms GC) and 3.34GB allocated for arithlst.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/ctblisoc.tst
    8609 ms (532 ms GC) and 1.58GB allocated for ctblisoc.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/ctblmoli.tst
   20422 ms (3585 ms GC) and 7.52GB allocated for ctblmoli.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/ctblsymm.tst
      78 ms (78 ms GC) and 1.16MB allocated for ctblsymm.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/direct_factors.tst
    2906 ms (249 ms GC) and 424MB allocated for direct_factors.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/grppcnrm.tst
   12782 ms (675 ms GC) and 1.52GB allocated for grppcnrm.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/grpprmcs.tst
    9062 ms (1838 ms GC) and 2.85GB allocated for grpprmcs.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/hash2.tst
    8875 ms (985 ms GC) and 1.18GB allocated for hash2.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/helptools.tst
     688 ms (109 ms GC) and 54.4MB allocated for helptools.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/hpc/alist.tst
#I  Test: File does not contain any tests!
       0 ms (0 ms GC) and 33.4KB allocated for hpc/alist.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/innerfunc.tst
    1453 ms (1107 ms GC) and 211MB allocated for innerfunc.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/matrix.tst
    3968 ms (248 ms GC) and 126MB allocated for matrix.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/AutomorphismGroup.tst
   47954 ms (6446 ms GC) and 14.4GB allocated for opers/AutomorphismGroup.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/ComplementClassesRepresentatives.tst
    6375 ms (720 ms GC) and 1013MB allocated for opers/ComplementClassesRepresentatives.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/IsomorphismGroups.tst
   14750 ms (1159 ms GC) and 3.39GB allocated for opers/IsomorphismGroups.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/Normalizer.tst
     812 ms (140 ms GC) and 132MB allocated for opers/Normalizer.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/SemidirectDecompositions.tst
    2234 ms (217 ms GC) and 354MB allocated for opers/SemidirectDecompositions.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/opers/StructureDescription.tst
    4844 ms (561 ms GC) and 760MB allocated for opers/StructureDescription.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/permgrp.tst
   55297 ms (7310 ms GC) and 14.7GB allocated for permgrp.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/processes/children.tst
       0 ms (0 ms GC) and 60.5KB allocated for processes/children.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/reesmat.tst
    5484 ms (250 ms GC) and 529MB allocated for reesmat.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/simplegrpit.tst
     188 ms (156 ms GC) and 1.64MB allocated for simplegrpit.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/sort.tst
    3234 ms (187 ms GC) and 183MB allocated for sort.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/stabchain.tst
     953 ms (156 ms GC) and 39.8MB allocated for stabchain.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/stablesort.tst
    9750 ms (312 ms GC) and 606MB allocated for stablesort.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/twocohom.tst
    8516 ms (297 ms GC) and 568MB allocated for twocohom.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/union.tst
    3828 ms (218 ms GC) and 842MB allocated for union.tst
testing: /home/pedro/tmp/gap-old/tst/teststandard/varnames.tst
     328 ms (172 ms GC) and 30.8MB allocated for varnames.tst
-----------------------------------
total    259797 ms (28657 ms GC) and 56.3GB allocated
              0 failures in 29 files

#I  No errors detected while testing

true

```

pedritomelenas commented:
`fail.log` contains 
Browse
carat
curlInterface-2.1.1
float-0.9.1
NormalizInterface-1.1.0
PolymakeInterface-2019.06.01
xgap-4.30
ZeroMQInterface-0.11

fingolfin commented:
Since parts of this PR were merged, I've now rebased it down to a single commit, which makes it much easier to see what is going on.

I also changed it to *not* switch to using the controversial `/cygdrive/c/....` style paths, as I realized that my core assumption was wrong: I thought that `TmpDirectory` returns such a path, and it is fine, so why not make `TmpName` work similarly. But while rebasing today, I discovered that `DirectoryTemporary` (the only caller of the kernel function `TmpDirectory`) actually checks for a `/cygdrive/?/` prefix and then transforms that into a `C:/...` path. That was added in a commit in March 2012, with this commit message:
> Temporarily fix DirectoryTemporary for Windows.
> It still returns something like
>   C:/WINDOWS/Temp/tmXXXXXX
> which is a newly created dir.
> We do not yet switch to backslashes, though, because this
> could lead to problems within GAP later...

So, I now went the opposite direction, and change `TmpDirectory` to directly produce a `C:/...` path, so that the GAP function `DirectoryTemporary` can be simplified again.

alex-konovalov commented:
I think before this merged I should test this under Windows properly, starting from `gap.bat` on a Cygwin-free machine.
